relayer/pkg/chainlink/txm/txm.go

```diff
@@ -211,20 +264,21 @@ func (txm *starktxm) confirmLoop() {
 			for addr := range hashes {
 				for i := range hashes[addr] {
 					hash := hashes[addr][i]
-					response, err := client.Provider.TransactionReceipt(ctx, caigotypes.StrToFelt(hashes[addr][i]))
+					f, err := starknetutils.HexToFelt(hash)
 					if err != nil {
-						txm.lggr.Errorw("failed to fetch transaction status", "hash", hash, "error", err)
+						txm.lggr.Errorw("invalid felt value", "hash", hash)
 						continue
 					}
-					receipt, ok := response.(caigorpc.InvokeTransactionReceipt)
-					if !ok {
-						txm.lggr.Errorw("wrong receipt type", "type", reflect.TypeOf(response))
+					response, err := client.Provider.GetTransactionStatus(ctx, f)
+					if err != nil {
+						txm.lggr.Errorw("failed to fetch transaction status", "hash", hash, "error", err)
 						continue
 					}
 
-					status := receipt.Status
+					status := response.FinalityStatus
```

Comment:
If it's REJECTED, then the tx won't be included in the nonce. So the ordering is wrong for subsequent txs... therefore we need to 

recalculate the current nonce value, resubmit subsequent transactions with re-calculated nonce

Reply:
Can you add a task for this to the next sprint?